app/components/autocomplete/at_mention/index.js

```diff
@@ -29,16 +51,34 @@ function mapStateToProps(state, ownProps) {
         }
     }
 
+    const match = postDraft.substring(0, cursorPosition).match(regex);
+    let matchTerm = null;
+    if (match) {
+        matchTerm = isSearch ? match[1] : match[2];
+    }
+
+    const opts = {
+        matchTerm,
+        currentUserId,
+        isSearch
+    };
+
+    if (isSearch) {
+        filterMembersInCurrentTeam(state, {...opts, array: teamMembers});
+    } else {
+        filterMembersInChannel(state, {...opts, array: inChannel});
+        filterMembersNotInChannel(state, {...opts, array: outChannel});
+    }
+
     return {
-        currentUserId: state.entities.users.currentUserId,
-        currentChannelId,
+        currentUserId,
+        currentChannelId: isSearch ? '' : currentChannelId,
```

Comment:
Instead of setting currentChannelId to be empty when we're searching, maybe handle that case in the component itself? It might be confusing later since you'd expect the currentChannelId to always be set